packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1070,8 +1071,8 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   ///
   /// After calling [dispose], the framework severs the [State] object's
   /// connection with the [BuildContext].
-  BuildContext get context => _element;
-  StatefulElement _element;
+  BuildContext? get context => _element;
```

Comment:
we should make this one non-nullable and make its getter generate a pretty FlutterError for the case where it's null, saying something like "This widget has been unmounted, so the State no longer has a context (and should be considered defunct). Consider canceling any active work during "dispose" or using the "mounted" getter to determine if the State is still active."

Reply:
Done.

Doing this change the test _Destination hero scrolls out of view mid-flight_ in `packages/flutter/test/widgets/heroes_test.dart` was failing. I fixed it with  https://github.com/flutter/flutter/pull/64672/commits/0cfceff260ce8ca98e0e4861c4ee28fbd4a43999#diff-b33b1120961b2c010e3388ede5bdf04aR493-R495. Not really sure if the fix is correct, could you take a look please?

Reply:
yeah that seems reasonable.